source/constructs/api/config/service.py

```diff
@@ -1,9 +1,46 @@
-from . import crud
+from . import crud,schemas
+import boto3
+from common.reference_parameter import admin_subnet_ids
 
 
 def set_config(key: str, value: str):
-    return crud.set_value(key, value)
+    crud.set_value(key, value)
+
+
+def get_config(key: str, default_value=None) -> str:
+    _value = crud.get_value(key)
+    if _value:
+        return _value
+    if default_value:
+        return default_value
+    return None
+
+
+def list_config():
+    return crud.list_config()
+
+
+def set_configs(configs: list[schemas.ConfigBase]):
+    for config in configs:
+        set_config(config.config_key, config.config_value)
+
+
+def list_subnets():
+    ec2_client = boto3.client('ec2')
+    response = ec2_client.describe_subnets(SubnetIds=admin_subnet_ids)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `describe_subnets` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)